pontoon/insights/tests/test_views.py

```diff
@@ -0,0 +1,237 @@
+import json
+from dataclasses import dataclass
+from datetime import datetime, timezone
+from http import HTTPStatus
+from unittest.mock import patch
+
+import pytest
+from dateutil.relativedelta import relativedelta
+from django.contrib.auth.models import User
+from django.shortcuts import render
+from django.urls import reverse
+
+from pontoon.actionlog.models import ActionLog
+from pontoon.insights import views
+from pontoon.test.factories import (
+    UserFactory,
+    ResourceFactory,
+    TranslationFactory,
+)
+
+
+def approve(translation, user, timestamp):
+    translation.approved = True
+    return perform_action(
+        ActionLog.ActionType.TRANSLATION_APPROVED, translation, user, timestamp
+    )
+
+
+def reject(translation, user, timestamp):
+    translation.approved = False
```

Comment:
Same as above in the `approve()` function.

Reply:
done